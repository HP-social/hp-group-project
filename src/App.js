import React, { Component } from 'react';
import { Provider } from 'react-redux';
import store from './ducks/store';
import { BrowserRouter } from 'react-router-dom';
import routes from './routes';
import Navigation from './components/Navigation/Navigation';
import './App.scss';

class App extends Component {
	render() {
		return (
			<Provider store={store}>
				<BrowserRouter>
					<div className='App'>
						{window.location.pathname === '/' ||
						window.location.pathname === '/sortinghat' ? null : (
							<Navigation />
						)}
						<div className='routes'>{routes}</div>
					</div>
				</BrowserRouter>
			</Provider>
		);
	}
}

export default App;
