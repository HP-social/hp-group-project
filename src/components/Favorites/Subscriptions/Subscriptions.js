import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Subscriptions.scss';
import HouseHeader from '../../Tools/HouseHeader/HouseHeader';
import axios from 'axios';

class Subscriptions extends Component {
	constructor(props) {
		super(props);
		this.state = {
			subscriptions: []
		};
	}

	componentDidMount() {
		this.getSubs();
	}

	getSubs = () => {
		axios
			.get(`/api/subscriptions/${this.props.user.wizard_id}`)
			.then((res) => this.setState({ subscriptions: res.data }));
	};

	deleteSubscription = (id) => {
		axios.delete(`/api/deletesubscription/${id}`).then(() => this.getSubs());
	};

	render() {
		let subsCards = this.state.subscriptions.map((e, i) => {
			return (
				<div className='subs_contents' key={i}>
					<div>
						<h1>{e.location}</h1>
					</div>
					<div className='inside_right'>
						<img
							onClick={() => this.deleteSubscription(e.forum_id)}
							src='https://s3.amazonaws.com/hp-project/unsubscribe24px.svg'
							alt='unsubscribe'
						/>
					</div>
				</div>
			);
		});

		return (
			<>
				<HouseHeader noButton='nope' house={''}>
					Subscriptions
				</HouseHeader>
				<div className='sub_outer' />
				<div className='subs_main'>{subsCards} </div>
			</>
		);
	}
}

function mapStateToProps(state) {
	const { user } = state;
	return {
		user
	};
}

export default connect(mapStateToProps)(Subscriptions);
