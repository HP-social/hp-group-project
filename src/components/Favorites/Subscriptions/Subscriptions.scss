.subs_main{
  background: lightgreen;
}