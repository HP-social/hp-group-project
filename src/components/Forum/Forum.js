import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setUser } from '../../ducks/reducer';
import axios from 'axios';
import './Forum.scss';
import Tweet from '../Tweet/Tweet';
import Card from './Cards/Card';

class Forum extends Component {
  constructor(props) {
    super(props);
    this.state = {
      post: [],
      makeATweet: false,
      user: {
        email: 'aestesc@gmail.com',
        wizard_id: 3,
        username: 'ronstoppable',
        house: 'slytherin',
        profile_img:
          'https://d36tnp772eyphs.cloudfront.net/blogs/1/2016/01/BENH9926-Edit.jpg'
      },
      forum: {}
    };
  }

  componentDidMount() {
    axios.get(`/api/forum/${this.props.match.params.id}`).then(result => {
      this.setState({ forum: result.data[0] });
    });
  }

  tweet = () => {
    this.setState({ makeATweet: !this.state.makeATweet });
  };

  render() {
    console.log(this.props);

    return (
      <div className='everything'>
        <h1 className='forum_title'>Slytherin Commons</h1>
        {/* <Tweet /> */}
        <div className='forum_card'>
          <Card />
          <Card />
          <Card />
          <Card />
          <Card />
          <Card />
          <Card />
          <Card />
        </div>
        {/* <Card /> */}
        {/* <div className='forum_post'>
          Ron is the best wizard
          <img id='wizard_avi'src={this.state.user.profile_img}></img>
          <div id='wizard_name'>{this.state.user.username}</div>
          <div id='time_of_post'>2 hours ago</div>
          <div id='star-five' />
          <div id='star-five_number'>39</div>
          <div id='comment' />
          <div id ='comment_number'>8</div>
          <div id='triangle' />
        </div> */}

        <button className='tweetButton' onClick={() => this.tweet()}>
          <img src='https://image.flaticon.com/icons/svg/1305/1305386.svg' />
        </button>
        {this.state.makeATweet === true ? (
          <Tweet newTweetStatus={this.tweet} />
        ) : null}
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { user } = state;
  return {
    user
  };
}

export default connect(
  mapStateToProps,
  { setUser }
)(Forum);
