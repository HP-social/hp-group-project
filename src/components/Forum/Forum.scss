@font-face {
	font-family: 'harryp';
	src: url('../../HARRYP__.TTF') format('truetype');
}

@font-face {
	font-family: 'lumos';
	src: url('../../LUMOS.TTF') format('truetype');
}

@font-face {
	font-family: 'parselto';
	src: url('../../PARSELTO.TTF') format('truetype');
}

$slytherin_green: #1a472a;
$slytherin_silver: #5d5d5d;
$gryffindor_red: #740001;
$gryffindor_gold: #d3a625;
$ravenclaw_blue: #0e1a40;
$ravenclaw_bronze: #946b2d;
$hufflepuff_black: #372e29;
$hufflepuff_yellow: #ecb939;

.everything {
	color: white;
	font-style: white;
	background: black;
	width: 100%;
	// height: 85.5vh;
	font-size: 2em;
	margin-top: 60px;
	//   margin-left: 100px;
	//   margin-right: 100px;
	display: flex;
	flex-direction: column;

	.forum_card {
		background: pink;
		height: 25%;
	}

	.forum_title {
		display: flex;
		background: green;
		justify-content: center;
		font-size: 50px;
		font-family: 'parselto';
		height: 10%;
	}
	.forum_post {
		height: 30%;
		display: flex;
		background: $slytherin_green;
		justify-content: center;
		align-content: center;
		align-items: center;
		border: 2px solid purple;
		position: relative;
		#wizard_name {
			position: absolute;
			left: 8vw;
			top: 2vh;
		}
		#wizard_avi {
			width: 75px;
			height: 75px;
			position: absolute;
			left: 1vw;
			top: 1vh;
			border-radius: 50%;
		}
		#time_of_post {
			background: $slytherin_green;
			opacity: 0.66;
			width: 150px;
			height: 30px;
			position: absolute;
			right: 2vw;
			top: 1vh;
			color: black;
			display: flex;
			justify-content: center;
			align-content: center;
			align-items: center;
		}
		#star-five {
			height: 20px;
			width: 20px;
			position: absolute;
			left: 2vw;
			bottom: 2vh;
			border-radius: 50%;
			border: 1px solid black;
		}
		#star-five:hover {
			background: red;
		}
		#star-five_number {
			position: absolute;
			left: 4.2vw;
			bottom: 2.5vh;
		}
		#comment {
			width: 20px;
			height: 20px;
			position: absolute;
			left: 8vw;
			bottom: 2vh;
			border: 1px solid black;
		}
		#comment:hover {
			background: paleturquoise;
		}
		#comment_number {
			position: absolute;
			left: 10vw;
			bottom: 2.5vh;
		}
		#triangle {
			width: 20px;
			height: 20px;
			border: 1px solid black;
			position: absolute;
			right: 2vw;
			bottom: 2vh;
		}
		#triangle:hover {
			background: yellow;
		}
	}
}
