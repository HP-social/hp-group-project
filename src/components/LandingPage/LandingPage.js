import React, { Component } from "react";
import { connect } from "react-redux";
import { setUser } from "../../ducks/reducer";
import "./LandingPage.scss";

class LandingPage extends Component {
  constructor() {
    super();
    this.state = {};
  }
  render() {
    return (
      <div className="main_landing">
        <div className="img_overlay" />
        <div className="top_castle" />
        <div className="mid_container">
					<div className='card'>1</div>
					<div className='card'>2</div>
					<div className='card'>3</div>
				</div>
        <div className="bottom_castle">hi3</div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { user } = state;
  return {
    user
  };
}

export default connect(
  mapStateToProps,
  { setUser }
)(LandingPage);
