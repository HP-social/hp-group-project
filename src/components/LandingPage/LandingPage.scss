.main_landing {
  text-align: center;
  height: 200vh;
  width: 100%;
  margin-left: -50px;

  .photo_before,
  .photo_after {
    background-attachment: fixed;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    height: 100%;
    width: auto;

    h1 {
      padding-top: 25%;
      font-size: 6em;
      color: white;
    }
  }

  // General icon style
  .scroll-icon {
    display: block;
    position: relative;
    height: 3em;
    width: 1.5em;
    border: 0.25em solid #fff;
    border-radius: 1em;
  }

  .scroll-icon__dot {
    display: block;
    position: absolute;
    left: 50%;
    background: #fff;
    height: 0.5em;
    width: 0.5em;
    top: 0.6em;
    margin-left: -0.25em;
    border-radius: 50%;
    transform-origin: top center;
    backface-visibility: hidden;
    animation: scroll_2 2s ease-out infinite;
  }

  // Example 2
  @keyframes scroll_2 {
    0% {
      transform: translateY(0);
      opacity: 1;
    }
    25% {
      opacity: 1;
    }
    75% {
      transform: translateY(1.2em);
      opacity: 0;
    }
    100% {
      transform: translateY(0);
      opacity: 0;
    }
  }

  .photo_before {
    background-image: url(https://s3.amazonaws.com/hp-project/Night_Hogwarts.jpg);
    -webkit-filter: grayscale(100%); /* Safari 6.0 - 9.0 */
    filter: grayscale(100%);
  }

  .photo_after {
    background-image: url(https://s3.amazonaws.com/hp-project/Night_Hogwarts.jpg);
    border-top: 1px solid #686868;
  }

  .mid_container {
    height: 60vh;
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    align-items: center;
    background: #9d9d9ba3;

    .card {
      display: flex;
      flex-direction: column;
      flex-wrap: wrap;
      justify-content: space-evenly;
      align-items: center;
      height: 40vh;
      width: 30vw;
      background: #f7f7f9b6;
      border-radius: 3%;
      img {
        height: 15vh;
        width: auto;
      }
      h1 {
        font-size: 2em;
        font-weight: bold;
        text-transform: uppercase;
        // padding: 1%;
      }
      p {
        font-size: 1.2em;
      }
    }
  }
  .bottom_container {
    height: 50%;
    background: #131213ce;
    background-image: url(https://images.unsplash.com/photo-1542270370-694e6a9a1f9e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80);
    background-position: center; /* Center the image */
    background-repeat: no-repeat; /* Do not repeat the image */
    background-size: cover; /* Resize the background image to cover the entire container */
    background-attachment: fixed; /* Parallax */

    h1 {
      padding-top: 15%;
      font-size: 6em;
      color: white;
    }
    h1:hover {
      cursor: pointer;
    }
  }
}
