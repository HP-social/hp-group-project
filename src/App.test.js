import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { checkPropTypes, string, array } from 'prop-types';
import configureStore from 'redux-mock-store';

//**** COMPONENTS TO BE TESTED ****
import LandingPage from './components/LandingPage/LandingPage';
import HouseHeader from './components/Tools/HouseHeader/HouseHeader';
import Forum from './components/Forum/Forum';
import Navigation from './components/Navigation/Navigation';
import BottomNav from './components/Navigation/BottomNav';
import Profile from './components/Profile/Profile';
import SortingHat from './components/Login/SortingHat/SortingHat';
import Subscriptions from './components/Favorites/Subscriptions/Subscriptions';

Enzyme.configure({ adapter: new Adapter() });

const initialState = {
  user: { house: 'gryffindor', username: 'patronuskiller', wizard_id: 2 },
  followingYou: { count: 0 }
};
const mockStore = configureStore();
let wrapper;
let store;
beforeEach(() => {
  store = mockStore(initialState);
});

describe('App renders', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('Landing Page Test', () => {
  test('Landing Page renders', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<LandingPage store={store} />);
    expect(wrapper.exists()).toBe(true);
  });

  test('props house is gryffindor', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<LandingPage store={store} />);
    expect(wrapper.props().user.house).toBe('gryffindor');
  });
});

describe('BottomNav state', () => {
  test('state should be string', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<BottomNav store={store} />);
    expect(wrapper.state('path')).toBe(undefined);
  });
});

test('house header props should be ravenclaw', () => {
  const store = mockStore(initialState);
  const wrapper = shallow(<HouseHeader store={store} house='ravenclaw' />);
  expect(wrapper.props().house).toEqual('ravenclaw');
});

test('Navigation component renders after user login', () => {
  const store = mockStore(initialState);
  const wrapper = shallow(<Navigation store={store} />);
  expect(wrapper.exists()).toBe(true);
});
test('username should be a dynamic string', () => {
  const store = mockStore(initialState);
  const wrapper = shallow(
    <Navigation store={store} username='patronuskiller' />
  );
  expect(wrapper.props().username).toEqual('patronuskiller');
});
test('Profile Comonent renders', () => {
  const store = mockStore(initialState);
  const wrapper = shallow(<Profile store={store} />);
  expect(wrapper.exists()).toBe(true);
});
test('Profile Comonent wizard ID should be 2', () => {
  const store = mockStore(initialState);
  const wrapper = shallow(<Profile store={store} />);
  expect(wrapper.props().user.wizard_id).toBe(2);
});

test('Profile Comonent followYou count should be 0', () => {
  const store = mockStore(initialState);
  const wrapper = shallow(<Profile store={store} />);
  expect(wrapper.props().followingYou.count).toBe(0);
});

//MOUNT
test('profile component following you count to be 0', () => {
  const store = mockStore(initialState);
  const wrapper = mount(<Profile store={store} />);
  expect(wrapper.props().followingYou.count).toBe(0);
});

describe('SortingHat', () => {
  it('displays an index less than 18 upon loading the page', () => {
    const store = mockStore(initialState);
    const sortingHat = shallow(<SortingHat store={store} />);
    expect(sortingHat.state('index')).toBe(undefined);
  });
});
describe('SortingHat', () => {
  it('has a sortingHatPicture', () => {
  const store = mockStore(initialState);
  const wrapper = mount(<SortingHat store={store} />);
  expect(wrapper.find('.sortingHatPicture').hasClass('sortingHatPicture')).to.equal(undefined);
});
});
