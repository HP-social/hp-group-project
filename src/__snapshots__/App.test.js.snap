// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SortingHat)
    debug={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "debug": true,
      "setUser": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "user": Object {
        "house": "gryffindor",
        "username": "patronuskiller",
        "wizard_id": 2,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "debug": true,
        "setUser": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "user": Object {
          "house": "gryffindor",
          "username": "patronuskiller",
          "wizard_id": 2,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`SortingHat should render banner text correctly with given strings 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SortingHat)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "setUser": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "user": Object {
        "house": "gryffindor",
        "username": "patronuskiller",
        "wizard_id": 2,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "setUser": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "user": Object {
          "house": "gryffindor",
          "username": "patronuskiller",
          "wizard_id": 2,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`SortingHat should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SortingHat)
    debug={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "debug": true,
      "setUser": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "user": Object {
        "house": "gryffindor",
        "username": "patronuskiller",
        "wizard_id": 2,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "debug": true,
        "setUser": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "user": Object {
          "house": "gryffindor",
          "username": "patronuskiller",
          "wizard_id": 2,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Tweet should render initial layout 1`] = `
Array [
  <Tweet
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "house": "gryffindor",
        "username": "patronuskiller",
        "wizard_id": 2,
      }
    }
  />,
]
`;

exports[`should render initial layout 1`] = `
Array [
  <Tweet
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    user={
      Object {
        "house": "gryffindor",
        "username": "patronuskiller",
        "wizard_id": 2,
      }
    }
  />,
]
`;
